src/libraries/System.Collections/src/System/Collections/Generic/SortedSet.cs

```diff
@@ -64,6 +64,28 @@ public partial class SortedSet<T> : ISet<T>, ICollection<T>, ICollection, IReadO
 
         internal const int StackAllocThreshold = 100;
 
+        // If Count is below or equal InternalIndexOfCountThreshold, InternalIndexOf returns a index
+        // referring to the position in a full tree:
+        //
+        //    0
+        //  1   2
+        // 3 4 5 6
+        // .......
+        //
+        // If Count is above InternalIndexOfCountThreshold,  it simply returns the index if it is enumerated
+        // Note that this is not used in SortedSet.TreeSubSet
+        //
+        // Some values for threshold, their respective max height and count of all indices of a full tree:
+        //
+        // | Threshold | MaxHeight | Full tree count
+        // |        30 |         8 |             511
+        // |        62 |        10 |            2047
+        // |       126 |        12 |            8191
+        // |       254 |        14 |           32767
+        // |       510 |        16 |          131071
+        // |      1022 |        18 |          524287
+        private const int InternalIndexOfCountThreshold = 254;
```

Comment:
Just took this value becuase it seemed "reasonable" to me according to numbers above in comments.